rtv/content.py

```diff
@@ -123,7 +125,14 @@ def strip_praw_submission(sub):
         data['subreddit'] = strip_subreddit_url(sub.permalink)
         data['flair'] = flair
         data['url_full'] = sub.url
-        data['url'] = ('selfpost' if is_selfpost(sub.url) else sub.url)
+
+        if reddit_link.match(sub.url):
+            stripped_url = reddit_link_no_host.search(sub.url).group()
+            stripped_comments = reddit_link_no_host.search(sub.permalink).group()
+            data['url'] = ('selfpost' if stripped_url == stripped_comments
+                    else 'x-post')
+        else:
+            data['url'] = sub.url
```

Comment:
I think this is a great way to  handle this. Having `data['url_type']` instead of combinations of other conditions will make the code much more readable. :+1: